app/services/service_result.rb

```diff
@@ -230,8 +230,9 @@ def initialize_errors(errors)
   end
 
   def new_errors_with_result
-    ActiveModel::Errors.new(self).tap do |errors|
-      errors.merge!(result) if result.try(:errors).present?
+    base = result.respond_to?(:errors) ? result : self
+    ActiveModel::Errors.new(base).tap do |errors|
+      errors.merge!(result) if base != self
```

Comment:
The method is effectively called in the initializer (via the `#initialize_errors` method). Because of that, it is order dependent. The result needs to be set before the method is called. Because of that, I would add the `result` as a parameter to this method so that the dependency becomes visible. 

Or the errors method is reworked to lazily initialize the object.